### PR TITLE
[21227] Utility to convert QoS ids and masks to string

### PR DESCRIPTION
**Pull request by Tempate**
This version includes `qos_policy_id_to_string` to convert a `QosPolicyId_t` to string and ` qos_policy_mask_to_string` to convert a `PolicyMask` to string.

### CODE REVIEW

**Comment by Tempate** [comment]
@mergifyio backport 0.x

File: cpp_utils/src/cpp/qos/qos_utils.cpp [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        if (qos_policy_mask.test(qos_policy_id))
+        {
+            policy_mask_str += ", " + qos_policy_id_to_string(qos_policy_id);
+        }
+    }
+
+    if (!policy_mask_str.empty())
+    {
+        policy_mask_str.erase(0, 2);
+    }
```
Comment (LuciaEchevarria99) [inline cpp_utils/src/cpp/qos/qos_utils.cpp:137]: I prefer not to delete the initial ' ,' from the string. While there's no perfect solution, one approach is to add a parameter to track whether the `QosPolicy` is the first in the sequence.

File: cpp_utils/test/unittest/qos/qos_utils_test.cpp [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+ * Test \c qos_policy_id_to_string call
+ */
+TEST(qos_utils_test, qos_policy_id_to_string)
+{
+    // User data
+    {
+        const auto id = QosPolicyId_t::USERDATA_QOS_POLICY_ID;
+        const auto id_str = utils::qos_policy_id_to_string(id);
+        const auto id_str_expected = "User data";
+        ASSERT_EQ(id_str, id_str_expected);
```
Comment (LuciaEchevarria99) [inline cpp_utils/test/unittest/qos/qos_utils_test.cpp:35]: Try to parametrize this test, you are writing the same steps to test all `QosPolicyId_t`, why not creating a `std::map<uint32_t, std::string>` (with every `QosPolicyId_t` and its corresponding `std::string`) and iterating through it?

**Review by LuciaEchevarria99 — CHANGES_REQUESTED** [review]

**Review by LuciaEchevarria99 — APPROVED** [review]
LGTM

**Pull request merged**